### PR TITLE
privileged: set mounts to rw

### PR DESCRIPTION
**Pull request by vrothberg**
Set all OCI mounts as privileged when running in privileged mode.

Fixes: #2625
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: server/container_create_linux.go [hunk 1]
```diff
@@ -626,6 +627,18 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID, contai
 		if err := specgen.AddOrReplaceLinuxNamespace(string(rspec.NetworkNamespace), netNsPath); err != nil {
 			return nil, err
 		}
+
+		if privileged {
+			specgen.RemoveMount("/sys")
+			specgen.RemoveMount("/sys/cgroup")
```
Comment (giuseppe) [inline server/container_create_linux.go]: it should be `/sys/fs/cgroup`

**Comment by vrothberg** [comment]
@giuseppe @rhatdan PTAL

File: server/container_create_linux.go [hunk 2]
```diff
@@ -686,6 +699,10 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID, contai
 		specgen.AddMount(mnt)
 	}
 
+	if privileged {
+		setOCIBindMountsPrivileged(&specgen)
```
Comment (giuseppe) [inline server/container_create_linux.go]: does it also override the mode for bind mounts that were configured as `ro`?  Probably doesn't matter as the privileged container can always remount it

Comment (vrothberg) [reply server/container_create_linux.go]: Those are added after that call, so we're good. Thanks for reviewing!

**Comment by vrothberg** [comment]
Preparing backports now.

**Pull request merged**